db-engine/src/main/java/com/splicemachine/db/iapi/types/SQLDecimal.java

```diff
@@ -1118,10 +1129,131 @@ private static int getWholeDigits(BigDecimal decimalValue) {
         if (ONE.compareTo(decimalValue) == 1)
             return 0;
         return decimalValue.precision() - decimalValue.scale();
-	}   
-	
+	}
+
 	public Format getFormat() {
 		return Format.DECIMAL;
 	}
 
+	/**
+	 *
+	 * Write to a Project Tungsten Format (UnsafeRow).
+	 *
+	 * @see UnsafeRowWriter#write(int, Decimal, int, int)
+	 *
+	 * @param unsafeRowWriter
+	 * @param ordinal
+     */
+	@Override
+	public void write(UnsafeRowWriter unsafeRowWriter, int ordinal) {
+		if (isNull())
+				unsafeRowWriter.setNullAt(ordinal);
+		else
+			unsafeRowWriter.write(ordinal, Decimal.apply(value),value.precision(),value.scale());
+	}
+
+	/**
+	 *
+	 * Read from a Project Tungsten Format (UnsafeRow).
+	 *
+	 * @see UnsafeRow#getDecimal(int, int, int)
+	 *
+	 * @param unsafeRow
+	 * @param ordinal
+	 * @throws StandardException
+     */
+	@Override
+	public void read(UnsafeRow unsafeRow, int ordinal) throws StandardException {
+		if (unsafeRow.isNullAt(ordinal))
+				setToNull();
+		else
+			value = unsafeRow.getDecimal(ordinal,precision,scale).toJavaBigDecimal();
+	}
+
+	/**
+	 *
+	 * TODO JL: This is not accurate...
+	 *
+	 * @return
+	 * @throws StandardException
+     */
+	@Override
+	public int encodedKeyLength() throws StandardException {
+		throw new RuntimeException("not implemented correctly");
+//		return isNull()?1:9;
```

Comment:
This appears to be the only data type that you removed; why remove this one and only this one?